docs/writing-docs/doc-blocks.md

```diff
@@ -0,0 +1,331 @@
+      table: {
+        type: { 
+            summary: 'something short', 
+            detail: 'something really really long' 
+        },
+      },
+      control: {
+        type: null,
+      },
+    },
+  },
+};
+```
+
+This would render a row with a modified description, a type display with a dropdown that shows the detail, and no control.
+
+If you find yourself writing the same definition over and over again, Storybook provides some convenient shorthands, that help you streamline your work.
+
+For instance you can use:
+
+- `number`, which is shorthand for `type:{name :'number'}`
+- `radio`, which is a shorhand for `control: {type: 'radio' }`
+
+##### MDX
+
+To customize argTypes in MDX, you can set an `mdx` prop on the `Meta` or `Story` components:
+
+```js
+
+<Meta
+  title="MyComponent"
+  component={MyComponent}
+  argTypes={{
+    label: { 
+        name: 'label',
+        /* other argtypes required */ 
+    },
+  }}
+/>
+
+<Story name="some story" argTypes={{
+  label: { 
+      name: 'different label', 
+      /* other required data */  
+    }
+}}>
+  {/* story contents */}
+</Story>
+```
+
+#### Controls
+
+The controls inside a props table are configured in exactly the same way as the [controls](../essentials/controls) addon pane. In fact you’ll probably notice the table is very similar! It uses the same component and mechanism behind the scenes.
+
+### Source
+
+Storybook Docs displays a story’s source code using the `Source` block. The snippet has built-in syntax highlighting and can be copied with the click of a button.
+
+![Docs blocks with source](./docblock-source.png)
+
+#### DocsPage
+
+In DocsPage, the `Source` block appears automatically within each story’s [Preview] block.
+
+To customize the source snippet that’s displayed for a story, set the `docs.source.code` parameter:
+
+```js
+
+export const CustomSource = () => Template.bind({});
+
+CustomSource.parameters = {
+  docs: { 
+      source: { 
+          code: 'Some custom string here';  
+      } 
+    },
+};
+```
+
+#### MDX
+
+You can also use the `Source` block in MDX. It accepts either a story ID or `code` snippet the `language` for syntax highlighting:
+
+```js
+import { Source } from '@storybook/addon-docs/blocks';
+import dedent from 'ts-dedent';
+
+<Source
+  language='css'
+  code={dedent`
+     .container {
+       display: grid | inline-grid;
+     }
+  `}
+/>
+```
+
+#### ⚒️ Description
+
+Storybook Docs shows a component’s description extracted from the source code or based on a user-provided string.
+
+![Docs blocks with description](./docblock-description.png)
+
+#### DocsPage
+
+<div style="background-color:#F8FAFC">
+TODO: waiting to be vetted
+</div>
+
+In DocsPage, a component’s description is shown at the top of the page. For [supported frameworks], the component description is automatically extracted from a docgen component above the component in its source code. It can also be set by the `docs.description` parameter.
```

Comment:
@jonniebigodes Add link supported frameworks page

Reply:
link is added